src/main/java/com/metamx/metrics/SigarUtil.java

```diff
@@ -46,6 +46,7 @@
         tmpDir.mkdir();
```

Comment:
we should use `Files.createTempDir()` instead of deleting the tempfile and creating a directory in-place.